src/S3/Transfer.php

```diff
@@ -340,7 +346,37 @@ private function getDownloadsIterator()
         return $this->source;
     }
 
-    private function upload($filename)
+    private function uploadAsNeeded($file)
+    {
+        if (!$this->overwrite) {
+            return $this->client
+                ->executeAsync($this->hasBeenUploadedCommand($file))
+                ->otherwise(function(Aws\S3\Exception\S3Exception $e) use ($file) {
+                    if ($e->getAwsErrorCode() === 'AccessDenied') {
+                        // The file does exist
+                        return;
```

Comment:
Should we be returning a value here to let a user know that the file already existed?

Reply:
We could but what do you think should be returned ? 

That makes me think about a different possibility : adding an explicit message to the debug output. And to do this, I think I can do the same as what is done in the "upload" functions : call the before option callback but this time with the HeadObject command. And then change a little bit the debugging message (for instance changing "Transferring" by "Ignoring" when HeadObject). What do you think ?